analyses/bangladesh/trigger_development/bgd-trigger.md

```diff
@@ -0,0 +1,569 @@
+```python
+from datetime import datetime
+
+import matplotlib.pyplot as plt
+import matplotlib as mpl
+from scipy.stats import norm, pearsonr
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+import read_in_data as rd
+from importlib import reload
+reload(rd)
+
+mpl.rcParams['figure.dpi'] = 300
+```
+
+### Create GloFAS objects
+
+```python
+da_glofas_reanalysis = rd.get_glofas_reanalysis()
+da_glofas_forecast = rd.get_glofas_forecast()
```

Comment:
Hmm I'm getting an error on this line...

```
AttributeError                            Traceback (most recent call last)
<ipython-input-2-b39b49459c1d> in <module>
      1 da_glofas_reanalysis = rd.get_glofas_reanalysis()
----> 2 da_glofas_forecast = rd.get_glofas_forecast()
      3 da_glofas_forecast_summary = rd.get_da_glofas_summary(da_glofas_forecast)
      4 da_glofas_reforecast = rd.get_glofas_reforecast()
      5 da_glofas_reforecast_summary = rd.get_da_glofas_summary(da_glofas_reforecast)

~\Desktop\pa-anticipatory-action\analyses\bangladesh\trigger_development\read_in_data.py in get_glofas_forecast()
     43     }
     44     da_glofas_forecast_dict = shift_dates(da_glofas_forecast_dict)
---> 45     return convert_dict_to_da(da_glofas_forecast_dict)
     46 
     47 

~\Desktop\pa-anticipatory-action\analyses\bangladesh\trigger_development\read_in_data.py in convert_dict_to_da(da_glofas_dict)
     99         dims=["leadtime_hour", "number", "time"],
    100         coords=dict(
--> 101             number=da_glofas_dict[120].number,
    102             time=time,
    103             leadtime_hour=list(da_glofas_dict.keys()),

~\anaconda3\envs\bang_floods\lib\site-packages\xarray\core\common.py in __getattr__(self, name)
    226                 with suppress(KeyError):
    227                     return source[name]
--> 228         raise AttributeError(
    229             "{!r} object has no attribute {!r}".format(type(self).__name__, name)
    230         )

AttributeError: 'DataArray' object has no attribute 'number'
```

Reply:
Hm that is strange. Are you using the latest data? It's referring to the following dataset:
`processed/bangladesh/GLOFAS_Data/bgd_cems-glofas-forecast_0120.nc`
If you open the data array directly, do you see that it has the variable `number`?

Reply:
Arg ok yes so when I'm opening ```processed/bangladesh/GLOFAS_Data/bgd_cems-glofas-forecast_v3_lt05d.nc``` I do see ```number```... I guess something about my xarray setup really doesn't like making things easy for me 😢

Reply:
Yeah this problem really sounds awful! I will message you with some more ideas on how to troubleshoot